launcher/src/main/java/org/apache/spark/launcher/AbstractCommandBuilder.java

```diff
@@ -241,7 +246,7 @@ List<String> buildClassPath(String appClassPath) throws IOException {
         }
       }
 
-      if (isRemote) {
+      if (isRemote || isBeeLine) {
```

Comment:
Does this affect spark-thrift-server which runs on spark classic mode?

Reply:
Should not.

What jars are required for BeeLine? `hive-*` jars (include `hive-jdbc`), all of those still present on the BeeLine's classpath.